### PR TITLE
Provide documentation for `DSharpPlus.Commands`

### PR DESCRIPTION
**Pull request by OoLunar**
# Summary
Provides articles and documentation for the `DSharpPlus.Commands` package. Additionally obsoletes both the `DSharpPlus.CommandsNext` and the `DSharpPlus.SlashCommands` packages.

# Notes
Untested.

### CODE REVIEW

File: docs/articles/commands/processors/slash_commands/choice_provider_vs_autocomplete.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+- Small Enums (Built-In support!)
+- Media types (e.g. `image`, `video`, `audio`)
+- The day of the week
+
+Some valid use-cases for autocomplete include:
+- Tag names
+- A Google search
+- Very large enums (e.g. all the countries in the world)
+
+## Implementing a Choice Provider
```
Comment (Plerx2493) [inline docs/articles/commands/processors/slash_commands/choice_provider_vs_autocomplete.md:44]: Maybe add information about DI for both implementations (injecting through ctor(?) and are they singletons?)

File: docs/articles/commands/custom_argument_converters.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+---
+uid: articles.commands.custom_argument_converters
+title: Custom Argument Converters
+---
+
+# Custom Argument Converters
```
Comment (Plerx2493) [inline docs/articles/commands/custom_argument_converters.md:6]: Here also: Do we provide DI for those?

File: docs/faq.md [hunk 3]
```diff
@@ -1,134 +1,139 @@
...
+#### Where are my pictures of spiderman?
+![GOD DAMN IT PETER][5]
+
+<!-- LINKS -->
+[0]: xref:articles.migration.3x_to_4x
+[1]: xref:articles.preamble
+[2]: https://dotnet.microsoft.com/download
+[3]: xref:articles.audio.voicenext.prerequisites
+[4]: xref:articles.beyond_basics.events
+[5]: xref:articles.commands_next.dependency_injection
```
Comment (Plerx2493) [inline docs/faq.md:134]: I think here was smth messed up

**Review by Plerx2493 — APPROVED** [review]
Some notes, otherwise looks good. Thank you for taking the time to write this all

**Comment by akiraveliara** [comment]
where did my section on parameter checks go :c

File: DSharpPlus.CommandsNext/CommandsNextExtension.cs [hunk 4]
```diff
@@ -22,6 +22,7 @@ namespace DSharpPlus.CommandsNext;
 /// <summary>
 /// This is the class which handles command registration, management, and execution.
 /// </summary>
+[Obsolete("DSharpPlus.CommandsNext is obsolete. Please consider using the new DSharpPlus.Commands extension instead.")]
```
Comment (akiraveliara) [inline DSharpPlus.CommandsNext/CommandsNextExtension.cs:25]: CommandsNext isn't fully replaceable yet, for lack of help and overloads

**Review by akiraveliara — CHANGES_REQUESTED** [review]

File: DSharpPlus.CommandsNext/ExtensionMethods.cs [hunk 5]
```diff
@@ -21,6 +21,7 @@ public static class ExtensionMethods
     /// <param name="client">Client to enable CommandsNext for.</param>
     /// <param name="cfg">CommandsNext configuration to use.</param>
     /// <returns>Created <see cref="CommandsNextExtension"/>.</returns>
+    [Obsolete("DSharpPlus.CommandsNext is obsolete. Please consider using the new DSharpPlus.Commands extension instead.")]
```
Comment (akiraveliara) [inline DSharpPlus.CommandsNext/ExtensionMethods.cs:24]: remove obsoletion here too

**Review by akiraveliara — CHANGES_REQUESTED** [review]

**Review by akiraveliara — APPROVED** [review]

**Pull request merged**